simdjson/src/simdjson.cpp

```diff
@@ -5591,7 +5591,6 @@ static inline uint32_t hex_to_u32_nocheck(
 // returns true if the provided byte value is a 
 // "continuing" UTF-8 value, that is, if it starts with
 // 0b10...
-[[maybe_unused]]
```

Comment:
Last time I was getting an `unused` warning during compilation for the function `is_utf8_continuing`. It seems that removing this line will bring the warning back?

Reply:
I didn't see a warning when I ran npm install, but if that's an issue then we'll want to fix it in core too :)

Reply:
> [...] but if that's an issue then we'll want to fix it in core too :)

Yeah, exactly, I forgot to open that issue.

----
Here on my machine I got that warning. It also didn't compile because `NApi` is complaining about the exceptions: 
```
../simdjson/bindings.cpp:114:3: error: cannot use 'try' with exceptions disabled
  try {
  ^
```

[Here](https://github.com/jkeiser/simdjson_nodejs/blob/update-api/binding.gyp#L23) is where I disable the exceptions. If I remove that line I get an error:
```
 Exception support not detected. Define either
      NAPI_CPP_EXCEPTIONS or NAPI_DISABLE_CPP_EXCEPTIONS.
```
And if I change it to `NAPI_CPP_EXCEPTIONS`, I get an error:
```
expanded from macro 'NAPI_THROW_IF_FAILED'
  if ((status) != napi_ok) throw Napi::Error::New(env);
```

I will have to read `NApi` documentation to check how we should handle exceptions.

Reply:
Ok, I found the crash, it only happens for XCode, so on XCode settings in the `binding.gyp` file we have to replace:

```
"xcode_settings": {
        "OTHER_CFLAGS": ["-std=c++17"],
 },
```
with
```
"xcode_settings": {
        "OTHER_CFLAGS": ["-std=c++17"],
        "GCC_ENABLE_CPP_EXCEPTIONS": "YES"
  },
```

Reply:
Have just fixed it. Thanks!